src/atom.coffee

```diff
@@ -783,11 +783,14 @@ class Atom extends Model
   showSaveDialog: (callback) ->
     callback(showSaveDialogSync())
 
-  showSaveDialogSync: (defaultPath) ->
-    defaultPath ?= @project?.getPaths()[0]
+  showSaveDialogSync: (options={}) ->
+    if _.isString(options)
+      options = defaultPath: options
     currentWindow = @getCurrentWindow()
     dialog = remote.require('dialog')
-    dialog.showSaveDialog currentWindow, {title: 'Save File', defaultPath}
+    options.title ?= 'Save File'
```

Comment:
Ok, is there an existing util method to do this… I see underscore has a shallow clone, but I don't to a deep clone. Currently (and seems reasonable for the future) the options will be JSON so if there's no existing util method I can just convert to JSON and back again to deep clone. What do you suggest?